File: doc/src/scripts/AttachmentModes.py
root = vessel.parts.root
stack = [(root, 0)]
while len(stack) > 0:
    part,depth = stack.pop()
    if part.axially_attached:
        attach_mode = 'axial'
    else: # radially_attached
        attach_mode = 'radial'
    print(' '*depth, part.title, '-', attach_mode)
    for child in part.children:
        stack.append((child, depth+1))

File: doc/src/scripts/AttachmentModes.lua
local root = vessel.parts.root
local stack = {{root, 0}}
while #stack > 0 do
  local part,depth = unpack(table.remove(stack))
  local attach_mode
  if part.axially_attached then
    attach_mode = 'axial'
  else -- radially_attached
    attach_mode = 'radial'
  end
  print(string.rep(' ', depth) .. part.title .. ' - ' .. attach_mode)
  for _,child in ipairs(part.children) do
    table.insert(stack, {child, depth+1})
  end
end
